ccm_web/server/src/api/api.course.handler.ts

```diff
@@ -118,10 +120,12 @@ export class CourseApiHandler {
   async mergeSections (sectionIds: number[]): Promise<CanvasCourseSectionBase[] | APIErrorData > {
     const NS_PER_SEC = BigInt(1e9)
     const start = process.hrtime.bigint()
-    const apiPromises = sectionIds.map(async (si) => {
-      const sectionHandler = new SectionApiHandler(this.requestor, si)
-      return await sectionHandler.mergeSection(this.courseId)
-    })
+    const apiPromises = createLimitedPromises<CanvasCourseSectionBase | APIErrorData>(
+      sectionIds.map(si => async () => {
+        const sectionHandler = new SectionApiHandler(this.requestor, si)
+        return await sectionHandler.mergeSection(this.courseId)
+      })
+    )
```

Comment:
I'd probably keep this here, since theoretically we allow up to 250 sections here, but the frontend isn't doing that (I think it's only ever one at a time, actually).

Reply:
I don't think we will ever merge 250 sections. Not a practical use case. I am fine if FE is not validating that and BE is is type validating/guarding it.

Reply:
Mmm, I was wrong about this. The "Go Merge" could trigger multiple unmerge calls at once. Unlikely to be very many, but the hard limit is 250. I suppose it's possible the user could hit that in the UI. Hopefully we have a way of representing that error.

Reply:
> Hopefully we have a way of representing that error.

I feel this should be addressed this as any other error during this API call. This case shouldn't be any exception. I think either way this issue is separate from this. You can create an separate issue but I won't necessary want this fix for initial release. since I am not anticipating people will ever merge 250 section but can be handled later.

Reply:
Yeah, I can do some testing as part of #272. It's an edge case though.